bigframes/display/anywidget.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from importlib import resources
+import functools
+import math
+from typing import Any, Dict, Iterator, TYPE_CHECKING
+import uuid
+
+import pandas as pd
+
+import bigframes
+
+ANYWIDGET_INSTALLED = True
+if TYPE_CHECKING:
+    import anywidget
+    import traitlets
+else:
+    try:
+        import anywidget
+        import traitlets
+    except Exception:
+        ANYWIDGET_INSTALLED = False
+
+
+class TableWidget(anywidget.AnyWidget):
```

Comment:
Presumably this line would fail if AnyWidget is not installed.

Maybe we can put the whole class definition inside an if block instead? Something like:

```python
if ANYWIDGET_INSTALLED:
    class TableWidget(anywidget.AnyWidget):
        ...
```

Alternatively, we could define a fallback for AnyWidget:

```python
if ANYWIDGET_INSTALLED:
    WIDGET_BASE = anywidget.AnyWidget
else:
    WIDGET_BASE = object


class TableWidget(WIDGET_BASE):
    ...
```